### PR TITLE
fix(openai): set stopReason in metadata so turns finish for Fireworks/Kimi providers (Fixes #1837)

### LINKED ISSUE #1837: Fireworks/Kimi provider: stopReason not set in metadata causes turns to never finish
## Bug Description

The OpenAI provider (used by Fireworks Kimi) is NOT setting \stopReason\ in IContent metadata when processing streaming responses. This causes the turn to never finish, making the AI appear \stuck\ or unresponsive after generating thinking blocks.

## Root Cause

The flow of data is broken at the OpenAI provider level:

1. OpenAIStreamProcessor captures \finish_reason\ from stream → state.lastFinishReason
2. OpenAIStreamProcessor yields IContent WITHOUT stopReason in metadata
3. MessageConverter.convertIContentToResponse() maps stopReason → finishReason on candidates
4. turn.ts checks resp.candidates?.[0]?.finishReason to yield \Finished\ event
5. Without finishReason, the \Finished\ event is never yielded
6. Without \Finished\ event, the turn never ends - causing the \stuck\ behavior

## Evidence

Comparing providers:
- Anthropic provider DOES set stopReason in metadata (AnthropicStreamProcessor.ts:473)
- OpenAI provider does NOT set stopReason in metadata

## Affected Code Locations

1. OpenAIStreamProcessor.ts:560-580 - combinedContent.metadata needs stopReason
2. OpenAIStreamProcessor.ts:600-620 - metadata-only response needs stopReason
3. OpenAINonStreamHandler.ts:65 - has finishReason but not setting stopReason

## Fix Required

Add stopReason to metadata in OpenAIStreamProcessor.ts when yielding IContent blocks. The finish_reason values from OpenAI (stop, length, tool_calls) should be mapped to the stopReason metadata field.

## Profile Configuration That Triggers This

- provider: Fireworks
- model: accounts/fireworks/routers/kimi-k2p5-turbo
- reasoning.enabled: true
- streaming: enabled

## Related

This explains why Fireworks Kimi often gets stuck and stops mid-conversation, especially when generating thinking blocks.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

The OpenAI provider (used by Fireworks, Kimi, and other OpenAI-compatible providers) was **not setting `stopReason` in IContent metadata** when processing streaming and non-streaming responses. This caused the turn to never finish, making the AI appear "stuck" or unresponsive after generating content.

## Root Cause

The data flow was broken at the OpenAI provider level:

1. `OpenAIStreamProcessor` captured `finish_reason` from stream chunks into `state.lastFinishReason`
2. `OpenAIStreamProcessor` yielded `IContent` **WITHOUT** `stopReason` in metadata
3. `MessageConverter.convertIContentToResponse()` maps `stopReason` → `finishReason` on candidates
4. `turn.ts` checks `resp.candidates?.[0]?.finishReason` to yield a `Finished` event
5. Without `finishReason`, the `Finished` event was **never yielded**
6. Without `Finished`, the turn never ended — causing the stuck behavior

### 1. OpenAIStreamProcessor.ts
- Added `mapFinishReasonToStopReason()` helper that maps OpenAI `finish_reason` values to the stopReason format expected by `MessageConverter`
- Set `stopReason` in metadata for **combined content yield** (thinking blocks + tool calls)
- Set `stopReason` in metadata for **metadata-only yield** (usage stats at end of stream)

### 2. OpenAINonStreamHandler.ts
- Added the same `mapFinishReasonToStopReason()` helper
- Set `stopReason` in metadata for **block content yield**
- Set `stopReason` in metadata for **metadata-only yield**
- Added fallback yield for stopReason-only when no blocks or usage

### 3. MessageConverter.ts
- Added OpenAI native `finish_reason` values (`stop`, `length`, `tool_calls`, `content_filter`) to the `finishReasonByStopReason` mapping for robustness

### Mapping: OpenAI → Internal

| OpenAI `finish_reason` | Internal `stopReason` | `FinishReason` |
|---|---|---|
| `stop` | `end_turn` | `STOP` |
| `length` | `max_tokens` | `MAX_TOKENS` |
| `tool_calls` | `tool_use` | `STOP` |
| `content_filter` | `end_turn` | `STOP` |

## Testing

Added `OpenAIStreamProcessor.stopReason.test.ts` with 15 behavioral tests covering:
- Streaming responses with stop/length/tool_calls finish reasons → correct finishReason
- Non-streaming responses → stopReason propagates correctly
- Thinking + text mixed responses → stopReason still propagates
- Edge cases (no stopReason, empty metadata, undefined metadata)
- OpenAI native finish_reason values in MessageConverter mapping

## Verification

- All 15 new tests pass
- All 532 existing OpenAI provider tests pass
- All 13 existing issue1729 stopReason tests pass (no regression)
- TypeScript compilation clean (no new type errors)
- ESLint clean (0 errors in modified files)
- Prettier formatted
- Build succeeds
- Smoke test passes

Fixes #1837

### CODE REVIEW

File: packages/core/src/providers/openai/OpenAINonStreamHandler.ts [hunk 1]
```diff
@@ -34,6 +34,24 @@ import {
...
+): string | undefined {
+  if (!finishReason) return undefined;
+  const mapping: Record<string, string> = {
+    stop: 'end_turn',
+    length: 'max_tokens',
+    tool_calls: 'tool_use',
+    content_filter: 'end_turn',
+  };
+  return mapping[finishReason] ?? finishReason;
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/core/src/providers/openai/OpenAINonStreamHandler.ts]: Addressed: extracted mapFinishReasonToStopReason into shared utility at finishReasonMapping.ts in 862c554be.

File: packages/core/src/providers/openai/OpenAIStreamProcessor.stopReason.test.ts [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+        speaker: 'ai',
+        blocks: [{ type: 'text', text: 'Filtered content' }],
+        metadata: {
+          stopReason: 'end_turn',
+        },
+      };
+
+      const response = geminiChat.convertIContentToResponse(icontent);
+      expect(response.candidates[0].finishReason).toBe('STOP');
+    });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/core/src/providers/openai/OpenAIStreamProcessor.stopReason.test.ts]: Addressed: renamed to 'should propagate end_turn stopReason through to finishReason STOP (no usage)' in 862c554be.

File: packages/core/src/providers/openai/OpenAIStreamProcessor.ts [hunk 3]
```diff
@@ -67,6 +67,24 @@ interface StreamingState {
...
+): string | undefined {
+  if (!finishReason) return undefined;
+  const mapping: Record<string, string> = {
+    stop: 'end_turn',
+    length: 'max_tokens',
+    tool_calls: 'tool_use',
+    content_filter: 'end_turn',
+  };
+  return mapping[finishReason] ?? finishReason;
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/core/src/providers/openai/OpenAIStreamProcessor.ts]: Addressed: same fix - both local copies removed, shared import from finishReasonMapping.ts in 862c554be.
